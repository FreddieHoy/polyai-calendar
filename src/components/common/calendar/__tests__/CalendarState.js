import { shallow } from 'enzyme';
import React from 'react';

import CalendarState from '../CalendarState';

describe('CalendarState', () => {
  it('renders the correct components', () => {
    const wrapper = shallow(<CalendarState />);
    wrapper.setState({ today: null });
    expect(wrapper).toMatchSnapshot();
  });

  it('adds one month to target date when switchMonth is called with e.target.id as plus', () => {
    const wrapper = shallow(<CalendarState />);
    const e = {
      target: {
        id: 'plus'
      }
    };
    wrapper.setState({ targetDate: new Date(2020, 0, 1) });
    wrapper.instance().switchMonth(e);

    const newTargetDate = wrapper.state('targetDate');
    expect(newTargetDate).toEqual(new Date(2020, 1, 1));
  });

  it('minus one month to target date when switchMonth is called with e.target.id as plus', () => {
    const wrapper = shallow(<CalendarState />);
    const e = {
      target: {
        id: ''
      }
    };
    wrapper.setState({ targetDate: new Date(2020, 2, 1) });
    wrapper.instance().switchMonth(e);

    const newTargetDate = wrapper.state('targetDate');
    expect(newTargetDate).toEqual(new Date(2020, 1, 1));
  });
});
