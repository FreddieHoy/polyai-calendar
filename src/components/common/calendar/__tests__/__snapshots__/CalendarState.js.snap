// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CalendarState renders the correct components 1`] = `
<styled.div>
  <MonthYear
    month="May"
    switchMonth={[Function]}
    year={2020}
  />
  <WeekDays />
  <Dates
    array={
      Array [
        "",
        "",
        "",
        "",
        "",
        "1",
        "2",
        "3",
        "4",
        "5",
        "6",
        "7",
        "8",
        "9",
        "10",
        "11",
        "12",
        "13",
        "14",
        "15",
        "16",
        "17",
        "18",
        "19",
        "20",
        "21",
        "22",
        "23",
        "24",
        "25",
        "26",
        "27",
        "28",
        "29",
        "30",
        "31",
        "",
        "",
        "",
        "",
        "",
        "",
      ]
    }
    today={null}
  />
</styled.div>
`;
